onnxruntime/core/mlas/inc/mlas.h

```diff
@@ -1410,17 +1420,45 @@ class MLAS_HALF_GEMM_POSTPROCESSOR {
     virtual ~MLAS_HALF_GEMM_POSTPROCESSOR() {}
 };
 
+#ifdef MLAS_F16VEC_INTRINSICS_SUPPORTED
+/**
+ * @brief Half precision activation functions
+*/
+class MLAS_HALF_GEMM_ACTIVATION_PROCESSOR : public MLAS_HALF_GEMM_POSTPROCESSOR {
+public:
+    MLAS_HALF_GEMM_ACTIVATION_PROCESSOR(
+        const MLAS_ACTIVATION& Activation
+        ) :
+            Activation_(Activation)
+    {}
+
+    void
+    Process(
+        MLAS_FP16* C,
+        size_t StartM,
+        size_t StartN,
+        size_t CountM,
+        size_t CountN,
+        size_t ldc
+        ) const override;
+
+private:
+    const MLAS_ACTIVATION& Activation_;
+};
+#endif // fp16 vector intrinsic supported
+
 /**
  * @brief Convert half gemm result matrix to single precision float matrix
 */
 class MLAS_HALF_GEMM_2FLOAT_PROCESSOR : public MLAS_HALF_GEMM_POSTPROCESSOR {
 public:
     MLAS_HALF_GEMM_2FLOAT_PROCESSOR(
+        const MLAS_ACTIVATION& Activation,
```

Comment:
>const MLAS_ACTIVATION& Activation, [](http://example.com/codeflow?start=8&length=34)

activation should be optional here

Reply:
and need to specify that this is an activation on fp32

Reply:
In the enum defining kind of activations, there's identity activation. The reason identify activation is there is to make the thing optional, right?

Why do we want to have a separate Activation for fp16? With no native fp16 datatype support in C/C++, the user has to specify in fp32 anyway right?